lib/takeaway.rb

```diff
@@ -0,0 +1,38 @@
+class Takeaway
+
+  attr_reader :basket
+
+  def initialize
+    @menu = {}
+    @basket = Hash.new(0)
+    @total = 0
+  end
+
+  def add_dish(name, price)
+    @menu.store(name, price)
+  end
+
+  def show_menu
+    @menu.to_s
+  end
+
+  def order(name, quantity = 1)
+    fail "Sorry, we don't have that!" unless @menu.has_key?(name)
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`. (https://github.com/bbatsov/ruby-style-guide#hash-key)